src/docfx/lib/JsonUtility.cs

```diff
@@ -79,20 +79,20 @@ public static string Serialize(object graph, Formatting formatting = Formatting.
         /// </summary>
         public static (List<Error>, T) Deserialize<T>(string json)
         {
-            var (errors, token) = Deserialize(json);
+            var (errors, token) = Deserialize(json, typeof(T));
             return (errors, (T)token.ToObject(typeof(T), DefaultDeserializer));
         }
 
         /// <summary>
         /// Parse a string to JToken.
         /// Validate null value during the process.
         /// </summary>
-        public static (List<Error>, JToken) Deserialize(string json)
+        public static (List<Error>, JToken) Deserialize(string json, Type type = null)
```

Comment:
Since we are traversing JToken for null value validation, I think we could do mismatching type validation at the same time, to avoid another traverse

---
In reply to: [205322205](https://github.com/dotnet/docfx/pull/3112#discussion_r205322205) [](ancestors = 205322205)

Reply:
Will separate the logic, and provide another JToken.ToObject<T> API

---
In reply to: [205351344](https://github.com/dotnet/docfx/pull/3112#discussion_r205351344) [](ancestors = 205351344,205322205)